### PR TITLE
Update Apache Log4j 2 to fix CVE-2021-44228

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Update Apache Log4j 2 to fix CVE-2021-44228

* Upgrade log4j2 to fix CVE-2021-44228
* Fix docs URL

See [JENKINS-67356](https://issues.jenkins.io/browse/JENKINS-67356)

### CODE REVIEW

**Comment by sunggun-yu** [comment]
Please merge and publish new version asap 😉

File: build.gradle [hunk 1]
```diff
@@ -110,7 +110,7 @@ jenkinsPlugin {
     displayName = 'Jenkins Checkmarx Plugin'
 
     // URL for plugin on Jenkins wiki or elsewhere
-    url = 'https://wiki.jenkins-ci.org/display/JENKINS/Checkmarx+CxSAST+Plugin'
+    url = 'https://plugins.jenkins.io/checkmarx/'
```
Comment (daniel-beck) [inline build.gradle:113]: plugins.jenkins.io as URL is always wrong: The plugin needs to point so somwhere else, so the plugins site can pick that up as a reference.

Comment (MarkEWaite) [reply build.gradle]: Thanks.  Reverted that change.  better to keep this pull request to a single commit.

**Review by Amir345 — APPROVED** [review]

**Comment by anitabee** [comment]
Any idea when this could be released?

**Comment by DuMaM** [comment]
@SubhadraSahoo could you merge this ? 🥺

**Comment by gowthamshankar99** [comment]
Can this be released asap please ? @MarkEWaite  @Amir345

**Comment by Amir345** [comment]
@gowthamshankar99 i don't have write access to the repo. sorry can't merge it :( waiting for the codeowners to merge it

**Review by gowthamshankar99 — APPROVED** [review]

**Comment by mattiassluis** [comment]
The plugin is managed by CheckMarx I believe. This is what we heard from them: 
> They respond that their teams are working to identify and analyze the impact of Apache Log4J Zero Day. We should have it by EOD today
Hopefully that moment will come soon.

**Comment by JuergenZimmer** [comment]
Hi all,
there is already an apache-log4j-2.16.0 available.
Best regards,
Juergen

**Comment by arunanc** [comment]
log4j 2.15.0 is [an incomplete fix](https://github.com/advisories/GHSA-7rjr-3q55-vv33). It's recommended to upgrade to 2.16.0.

**Comment by Wadeck** [comment]
⚠️ Please update to 2.16 to also fix the new one: https://nvd.nist.gov/vuln/detail/CVE-2021-45046

File: build.gradle [hunk 2]
```diff
@@ -60,9 +60,9 @@ dependencies {
             'com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.11.3',
             'com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.10.5',
             'com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.10.5',
-            'org.apache.logging.log4j:log4j-slf4j-impl:2.13.3',
-            'org.apache.logging.log4j:log4j-api:2.13.3',
-            'org.apache.logging.log4j:log4j-core:2.13.3'
+            'org.apache.logging.log4j:log4j-slf4j-impl:2.15.0',
```
Comment (mwebbelsevier) [inline build.gradle:63]: probably needs to be 2.16.0 as of this morning

File: build.gradle [hunk 3]
```diff
@@ -60,9 +60,9 @@ dependencies {
...
+            'org.apache.logging.log4j:log4j-api:2.15.0',
+            'org.apache.logging.log4j:log4j-core:2.15.0'
```
Comment (MichaelSp) [inline build.gradle:65]:
```suggestion
            'org.apache.logging.log4j:log4j-slf4j-impl:2.16.0',
            'org.apache.logging.log4j:log4j-api:2.16.0',
            'org.apache.logging.log4j:log4j-core:2.16.0'
```

**Comment by scurvydoggo** [comment]
My company uses Checkmarx and I have been told that they will be releasing this on the 24th. Ostensibly because their focus is on their 'cxflow' tool is the priority, however they only need press merge butan :/

**Comment by MarkEWaite** [comment]
Superseded by #81 .  Thanks for merging the change

**Comment by m1madness** [comment]
apache log4j recommends version 2.17. 2.16 is still vulnerable. https://logging.apache.org/log4j/2.x/security.html

**Pull request closed without merging**